lib/routes/account.js

```diff
@@ -1179,19 +1179,20 @@ module.exports = function (
                */
               return db.verifyTokens(code, account)
                 .catch(function (err) {
-                  if (err.errno === error.ERRNO.INVALID_VERIFICATION_CODE && butil.buffersAreEqual(request.payload.code, account.emailCode)) {
+                  if (err.errno === error.ERRNO.INVALID_VERIFICATION_CODE && butil.buffersAreEqual(code, account.emailCode)) {
                     // The code is just for the account, not for any sessions
                     return true
-                  } else {
-                    throw error.invalidVerificationCode()
                   }
+                  throw err
```

Comment:
I think it's worth preserving the original error here, in case it's e.g. an operational db error